src/cljdoc/doc_tree.clj

```diff
@@ -139,29 +169,24 @@
 
 ;; Deriving doctrees -----------------------------------------------------------
 
-(defn- supported-file-type [path]
-  (cond
-    (.endsWith path ".markdown") :markdown
-    (.endsWith path ".md")       :markdown
-    (.endsWith path ".adoc")     :asciidoc))
-
 (defn- readme? [path]
   (and (.startsWith (.toLowerCase path) "readme.")
-       (supported-file-type path)))
+       (try (filepath->type path) (catch Exception _ false))))
 
 (defn- changelog? [path]
   (and (some #(.startsWith (.toLowerCase path) %) ["changelog." "changes."  "history." "news." "releases."])
-       (supported-file-type path)))
+       (try (filepath->type path) (catch Exception _ false))))
 
 (defn- doc? [path]
-  (and (supported-file-type path)
-       (or (.startsWith path "doc/")
-           (.startsWith path "docs/"))))
+  (and (or (.startsWith path "doc/")
+           (.startsWith path "docs/"))
+       (try (filepath->type path) (catch Exception _ false))))
 
 (defn- infer-title [path file-contents]
-  (or (case (supported-file-type path)
-        :markdown (second (re-find #"(?m)^\s*#+\s*(.*)\s*$" file-contents))
-        :asciidoc (second (re-find #"(?m)^\s*=+\s*(.*)\s*$" file-contents)))
+  (or (case (filepath->type path)
+        ;; NOTE infer-title will fail with non adoc/md files
+        :cljdoc/markdown (second (re-find #"(?m)^\s*#+\s*(.*)\s*$" file-contents))
+        :cljdoc/asciidoc (second (re-find #"(?m)^\s*=+\s*(.*)\s*$" file-contents)))
```

Comment:
Could this also benefit from a multimethod?

Reply:
It could but I don't want to put a lot more stuff into the infer-logic, people should use `doc/cljdoc.edn` and cljdoc should provide appropriate tooling to generate those files, verify them, etc.